### DISCUSSION TITLE
26일 미팅 추가 정보 전달 및 아이디어 토론

### DISCUSSION THREAD

**Discussion by JO-HEEJIN** [birth2death/research_ai#30] [Ideas]
1. 현재는 1,2,3,4 단계로 계단식으로 3d object가 일정 시간이 되면 변경되는 로직을 가지고 있다. 
2. 만약 각 단계에서 변화 될 때 메쉬가 조금씩 다음 단계 object와 비슷하게 성장하도록 메시 변형이 가능한 지
3. 현재 렌더러가 필요한 이유는 각 오브젝트가 변환될 때 프레임이 틔기 때문이다. 그 프레임이 변화되는 사이 파티클 에미션을 넣어서 오브젝트를 숨겨야 하기 때문, 그런데 현재 애플 엔진에서는 아직 emission이 지원이 안 된다. 왜냐하면 렌더러가 블렌더를 비롯 다른 엔진과 다르기 때문
4. 현재 그 분야 렌더러로 최고봉은 언리얼, 그러나 언리얼은 지원이 안 되는 상태 그러기에 결국 애플의 grapic API를 건드려야 하는 상황
5. metal 을 응용해서 쉐이더로 최대한의 fluffy 함을 살려야 하는데 역시나 이 부분은 애플이 개발하기엔 아직 이름. 약 5년은 기다려야 함.
6. 그러나 언리얼 api 를 응용하면 가능성이 있음.

**Comment by binarydaddy** [comment]
**현재 기획상 AI가 왜 필요한가?**

(1) Task의 성격에 맞춰 무궁무진하게 다양한 캐릭터를 만들어주고 싶다.

- 우울한 분위기의 Task가 등록되어있다면 힘이 나는 캐릭터를 만들어주고, 쉬운일이 들어왔을때는 조금 라이트한 느낌이 나는 참새를 만들어주고 등등. (분위기 파악: LLM 활용, Character generation: diffusion / nerf 활용)

(2) 단가를 낮출 수 있다

- 사실상 아티스트만 많이 들어간다고 하면 AI가 필요없음.

---

(2)은 사실 설득력이 없을것. 다른 덩치 큰 회사가 기획 아이템 그대로 써서 3D 캐릭터만 찍어내면 끝이니까.
(1)이 좀 두터운 진입장벽을 만들어 줄 수 있을 것이라 생각됨.

---

(1)의 서비스 Flow

Task 생성할때마다 task의 특성이 반영된 새로운 캐릭터가 하나씩 만들어짐.

- Template 기반으로 variation이 다른 애들이 생성됨 (공룡타입, 새타입 등 몇가지 정해진 template에 디테일 (살집, 뿔, 눈크기 등)이 추가된 형태).
- Variation은 분위기에 따라 랜덤으로 그떄그때 서버에서 생성해서 client로 보내줌.

Task가 진행됨에 따라 늙어감.

- Task 기간이 끝나면 만들어졌던 캐릭터는 다시 만날 방법이 없음.
- 다시 만나고 싶으면 프리셋 구매 (결제 유도).
- 반대로 새로운 캐릭터를 뽑고싶으면 결제를 유도하는것이 좋으나, 현재 많은 2D diffusion BM이 그렇게 되어있고, 소비자의 입장에서 반감이 들때가 있음.

(1)에 대해 좀 더 깊게 생각을 해보면,
엄청나게 많은 종류의 동물을 만들어주면 물론 좋겠지만, animation / mesh detail이 보존이 잘 안될것임.
정해진 몇 종류만 일단 만들어주되, variation을 굉장히 많이 줄 수 있으면 사실상 무궁무진하게 많은 캐릭터라고 보일 수 있지 않을까라고 생각함. (Variation에는 단순 texture를 바꿔줄수도 있지만, 뿔이 달리거나, 살이 더 붙거나 등등도 가능할 것으로 생각).

추가적으로, template을 생성된 3D object에 맞추는 형식으로 생성을 하게 되면, vertex topology가 공유되기 때문에 blendshape을 활용할 수 있음. → 구현적인 이점이 존재 (오브젝트 바꾸기 위해 해결해야 하는 문제들보다 훨씬 쉬울 것).

**Comment by binarydaddy** [comment]
1,2,3,4 3d object를 모두 동일한 topology로 맞추어 놓고 아티스트에게 작업을 의뢰.
Texture / Material / Vertex 위치 모두를 time축에 대한 blendshape으로 만들어놓고 linear하게 풀면 자연스러운 변화를 만들 수 있을것이라 생각함.
